packages/edit-site/src/components/global-styles/font-library-modal/font-collection.js

```diff
@@ -77,36 +75,27 @@ function FontCollection( { slug } ) {
 				await getFontCollection( slug );
 				resetFilters();
 			} catch ( e ) {
-				setNotice( {
-					type: 'error',
-					message: e?.message,
-					duration: 0, // Don't auto-hide.
-				} );
+				if ( ! notice ) {
```

Comment:
Do we need to check for notice?

Reply:
I added this as the `fetchFontCollection` function keeps re-setting the notice on failure if there is an error with the font collection (e.g. something wrong with the JSON file), and it was causing an infinite loop of setting the error notice, I think because `setNotice` is a dependency of the `useEffect` where this function is called. This line fixed the issue (original discussion [here](https://github.com/WordPress/gutenberg/pull/58180#issuecomment-1910358720)).